src/xscen/io.py

```diff
@@ -272,22 +248,42 @@ def clean_incomplete(path: str | os.PathLike, complete: Sequence[str]) -> None:
     ----------
     path : str, Path
         A path to a zarr folder.
-    complete : sequence of strings
-        Name of variables that were completed.
+    complete : sequence of strings, optional
+        Name of variables that were completed. All other variables (except coordinates) will be removed.
+        Use either `complete` or `incomplete`, not both.
+    incomplete : sequence of strings, optional
+        Name of variables that should be removed.
+        Use either `complete` or `incomplete`, not both.
 
     Returns
     -------
     None
     """
     path = Path(path)
-    with xr.open_zarr(path) as ds:
-        complete = set(complete).union(ds.coords.keys())
 
-    for fold in filter(lambda p: p.is_dir(), path.iterdir()):
-        if fold.name not in complete:
-            msg = f"Removing {fold} from disk"
-            logger.warning(msg)
-            sh.rmtree(fold)
+    if complete is not None and incomplete is not None:
+        raise ValueError("Use either `complete` or `incomplete`, not both.")
+
+    if complete is not None:
+        with xr.open_zarr(path) as ds:
+            complete = set(complete).union(ds.coords.keys())
+
+        for fold in filter(lambda p: p.is_dir(), path.iterdir()):
+            if fold.name not in complete:
+                msg = f"Removing {fold} from disk"
+                logger.warning(msg)
+                sh.rmtree(fold)
+
+    elif incomplete is not None:
+        for fold in filter(lambda p: p.is_dir(), path.iterdir()):
+            if fold.name in incomplete:
+                msg = f"Removing {fold} from disk"
+                logger.warning(msg)
+                sh.rmtree(fold)
+
+    # Remove .zmetadata to avoid issues with zarr and xarray
+    if (path / ".zmetadata").exists():
+        Path.unlink(path / ".zmetadata")
```

Comment:
when the calculation that failed is redone, will it add a new complete zmetadata ?
why don't we just regenerate a correct one here?